app/models/user.rb

```diff
@@ -9,9 +9,67 @@ class User < ApplicationRecord
   validates :first_name, :last_name, :username, presence: true
   validates :username, uniqueness: true
 
-  def data
-    User.all.as_json.find do |user|
-      user['username'] == username
+  has_many :active_friendships, foreign_key: :active_friend_id, dependent: :destroy, class_name: 'Friendship'
+  has_many :passive_friendships, foreign_key: :passive_friend_id, dependent: :destroy, class_name: 'Friendship'
+  has_many :active_friends, through: :passive_friendships, source: :active_friend, dependent: :destroy
+  has_many :passive_friends, through: :active_friendships, source: :passive_friend, dependent: :destroy
+
+  default_scope { order(created_at: :asc) }
+
+  def friends
+    User.where(id: active_friendships.where(confirmed: true).pluck(:passive_friend_id))
+      .or(User.where(id: passive_friendships.where(confirmed: true).pluck(:active_friend_id)))
+  end
+
+  def pending_received_requests_from
+    User.where(
+      id: passive_friendships
+        .where(confirmed: false)
+        .pluck(:active_friend_id)
+    )
+  end
+
+  def pending_sent_requests_to
+    User.where(
+      id: active_friendships
+        .where(confirmed: false)
+        .pluck(:passive_friend_id)
+    )
+  end
+
+  def mutual_friends_with(other_user, page)
+    offset = (page.to_i - 1) * 10
```

Comment:
10 could be saved in constant called "PER_PAGE", easy to change if you need bigger pages at some point

Reply:
You're right! Ok what I did now is just passed this `per_page` as an argument. And in `friends_controller` and `mutual_friends_controller` I just included a variable for it.

Kindly see updated `models/user.rb`, `controllers/v1/users/friends_controller.rb` and `controllers/v1/users/mutual_friends.rb` for these changes :)